logger.go

```diff
@@ -5,155 +5,260 @@ import (
 	"os"
 	"runtime"
 	"strings"
+	"sync"
+	"testing"
 	"time"
 )
 
-type logWrappedT struct {
-	T
+type loggerT[ET T] struct {
+	runTHelper    // Embeds T and provides Fail/Parallel
+	orig       ET // Keep reference to original for Run method
```

Comment:
I'm wondering why an underlying `T` is embedded in `runTHelper` and referenced as `orig`.  The comment says `orig` is for the "Run method"... but if `runTHelper` can't run, then why is "run" in the name?